app/controllers/courses_controller.rb

```diff
@@ -49,6 +49,12 @@ def update
     ensure_passcode_set
     UpdateCourseWorker.schedule_edits(course: @course, editing_user: current_user)
     render json: { course: @course }
+  rescue Wiki::InvalidWikiError => e
+    language = params.dig(:course, :home_wiki, :language)
```

Comment:
Hmm... doing it this way will result in an incorrect error message if, rather the the home wiki, the invalid wiki comes from the set of additional tracked wikis. Rather than digging for the parameter from the request, I think the correct behavior should be to get the invalid domain from the error object. (This might require a change to how the error in generated in wiki.rb.)

Reply:
Hmm. So you mean i should pass incorrect domain through the error obj in wiki.rb and should try to fetch that obj in here.?

Reply:
Yes. The error object is available already (`e`) but I'm not sure whether it currently includes the project and language for the invalid wiki.

Reply:
> Yes. The error object is available already (`e`) but I'm not sure whether it currently includes the project and language for the invalid wiki.

@ragesoss I've Done this please check my approach once. is this what you were talking about?